x-pack/plugins/apm/server/lib/metrics/by_agent/java/heap_memory/index.ts

```diff
@@ -49,6 +47,9 @@ const chartBase: ChartBase<HeapMemoryMetrics> = {
 };
 
 export async function getHeapMemoryChart(setup: Setup, serviceName: string) {
-  const result = await fetch(setup, serviceName);
-  return transformDataToMetricsChart<HeapMemoryMetrics>(result, chartBase);
+  const result = (await fetch(setup, serviceName)) as MetricSearchResponse<
+    keyof typeof chartBase.series
+  >;
+
+  return transformDataToMetricsChart(result, chartBase as ChartBase);
```

Comment:
Yeah definitely looks better! I tried to make this work but ran into some issues when casting the return value to the return type. I've landed on a solution where instead of having several `fetcher`s and `fetch` calls, we have a generic `fetchAndTransformMetrics` call with some parameters. `fetchAndTransformMetrics` then calls `transformDataToMetricsChart`, and TypeScript then - for some reason - is able to match the types together. Bonus is that we can merge a bunch of largely overlapping `fetcher` files into one call.